internal/pkg/deploy/cloudformation/env.go

```diff
@@ -64,6 +64,8 @@ func (cf CloudFormation) UpdateAndRenderEnvironment(conf StackConfiguration, buc
 		}
 		return changeSetID, nil
 	}
+	in.enableInterrupt = true
+	in.detach = true
```

Comment:
Why do we always set them to true for env deploy, which is not the same as https://github.com/aws/copilot-cli/pull/5123/files#diff-082bb2a58e8378e094bbe9302a375183abd32e5cce67e2e4f90bc46fd5302c41R30?

Reply:
This is because function [`func (cf CloudFormation) newUpsertChangeSetInput`](https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/deploy/cloudformation/cloudformation.go#L311) is getting called in both 
[`func (cf CloudFormation) DeployService`](https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/deploy/cloudformation/workload.go#L18) and [`func (cf CloudFormation) DeployTask`](https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/deploy/cloudformation/task.go#L20)

But we want to enable interrupt and detach for `workloads` not for `task run`.

Reply:
but isn't `detach` the param defined but not used at all?

Reply:
oops sorry it should be `in.detach=detach`